File: check_bbox.py
import firebase_admin
from firebase_admin import credentials, firestore

cred = credentials.ApplicationDefault()
firebase_admin.initialize_app(cred, {'projectId': 'stitch-dev-3'})
db = firestore.client()

chunks_ref = db.collection('tenants').document('default').collection('documents').document('doc_006').collection('chunks')
docs = chunks_ref.where('page_number', '==', 7).get()

for doc in docs:
    data = doc.to_dict()
    text = data.get('text', '')[:100].replace('\n', ' ')
    print(f"Chunk ID: {doc.id}")
    print(f"Text preview: {text}")
    print(f"Bbox: {data.get('bbox')}")
    print('---')

File: check_bbox.js
const { Firestore } = require('@google-cloud/firestore');
const firestore = new Firestore();

async function check() {
  const chunksRef = firestore.collection('tenants').doc('default').collection('documents').doc('doc_006').collection('chunks');
  const snapshot = await chunksRef.where('page_number', '==', 7).get();
  snapshot.forEach(doc => {
    const data = doc.data();
    console.log(`Chunk ID: ${doc.id}`);
    console.log(`Text preview: ${data.text.substring(0, 100).replace(/\n/g, ' ')}`);
    console.log(`Bbox: ${JSON.stringify(data.bbox)}`);
    console.log('---');
  });
}
check().catch(console.error);
